src/data/int/gcd.lean

```diff
@@ -1,85 +1,28 @@
 /-
 Copyright (c) 2018 Guy Leroy. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Sangwoo Jo (aka Jason), Guy Leroy, Johannes Hölzl, Mario Carneiro
+Authors: Sangwoo Jo (aka Jason), Guy Leroy, Johannes Hölzl, Mario Carneiro, Aaron Anderson
 -/
-import data.nat.prime
+
+import data.int.xgcd
+import ring_theory.multiplicity
+import ring_theory.principal_ideal_domain
```

Comment:
This is a lot of ring theory to import for a file in `data` that gets imported by a lot of other files. If a maintainer thinks that's ok, then this whole PR looks good to me.

Reply:
(Except for the merge conflicts, sorry about that)

Reply:
@paulvanwamelen what do you think of starting `ring_theory/int/gcd.lean`?
I agree that it's a good idea to separate the "specialised" lemmas about `nat` and `int` from the general ones. But I also agree with @awainverse that we shouldn't have "heavy duty" imports in this "low-level" file.

Reply:
So the idea being that `data/int/gcd.lean` would have results on `int.gcd` that don't require (or currently use) ring theory
while `ring_theory/int/gcd.lean` would contain results on `int.gcd` that do require (or currently use) ring theory, right?
That works for me, but where do we put e.g. 
```
theorem lcm_comm (i j : ℤ) : lcm i j = lcm j i :=
by simp [(int.coe_nat_eq_coe_nat_iff _ _).symm, coe_lcm, lcm_comm]
```
It uses ring theory, but clearly doesn't need to. Should we reprove these without importing ring theory? It doesn't seem right having `gcd_comm` in `data/int/gcd.lean`, but `lcm_comm` in `ring_theory/int/gcd.lean`. On the other hand it seems right to prove the special case from the general theory. Maybe that is why things are the way they are now, and we should just leave things as they are.

Reply:
Good question. The reasoning is that I can imagine that some tactics for speeding up computations might want to import `data/int/gcd`. These tactics should then become available as early as possible, so they shouldn't import a large part of theory.
I therefore propose that either (i) `lcm_comm` and `gcd_comm` are not in the same file, or (ii) you reprove the special.without using general theory.

Reply:
Thanks Johan, I'll refactor a bit more then!